File: JAVA GITS ASSINGMENT/2026 project/FastingWeekend.java
public class FastingWeekend{
    public static void main(String[] args){

//1
String name = "Rosemary";
String reverse = "";
for(int index = name.length() -1; index >= 0; index--){
reverse += name.charAt(index);
}
System.out.println("reversed of Rosemary: " + reverse);
System.out.println();


//2
String nums = "1222211";
String rev = " ";
for(int in = nums.length() -1; in >= 0; in--){
rev += nums.charAt(in);
}
System.out.println("reversed of Rosemary: " + rev);
System.out.println();



//3
String upper = "Faith";
int counter = 0;
for(int low = 0; low < upper.length(); low++){
       char cased = upper.charAt(low);
    if(Character.isUpperCase(cased)){
        counter ++;

    }
}
        System.out.println("Number of uppercase = " + counter);
System.out.println();
    


//4
String lower = "Faith";
int count = 0;
for(int low = 0; low < lower.length(); low++){
       char cases = lower.charAt(low);
    if(Character.isLowerCase(cases)){
        count ++;

    }
}
        System.out.println("Number of lowercase = " + count);
System.out.println();
    


//6
String named = "rosemary";
for(int index = 0; index < named.length(); index++){
System.out.println((int) named.charAt(index));

}
System.out.println();



//7
double sum = 0;
for(double num = 1; num <=100; num ++){
sum += num; 
}
double average = sum / 100;
System.out.println("Average of 1 - 100 =" + average);

System.out.println();




//8
int divisor_number = 12;
for(int con = 1; con < 13; con++){
    if(divisor_number % con == 0){

        System.out.print("Numbers divisible by 12 =" + con);
System.out.println();

}
}


//9
int divi_number = 12;
int total = 0;
for(int cont = 1; cont < 13; cont++){
    if(divi_number % cont == 0){
    total += cont;
}
}
        System.out.print("Total divisors = " + total);
System.out.println();




//10

//11





//14

int totalSum = 0;
for(int even = 1; even <= 100; even ++){
    if(even % 2 == 0){
     totalSum += even;
}
}
System.out.print("Sum of even digits from 1-100 = " + totalSum);
System.out.println();


//15

int totals = 0;
for(int odd = 1; odd <= 100; odd ++){
    if(odd % 3 == 0){
     totals += odd;
}
}
System.out.print("Sum of odd digits from 1-100 = " + totals);
System.out.println();
















    }




}

File: Phython GITS assignment /Python 2026/FastingWeekend.py
from math import sqrt

#1
name = "rosemary"
for names in name:
    print(f"Reverse string of {name} =", name[::-1], end = " ")
    break
print()


#2
intt = 1234
print(f"Reverse int of {intt} =", str(intt)[::-1])


#6

name = "faith"
for number in name:
    print(ord(number), end = " ")


print()

#7
sums = 0

for numbers in range(1, 101):
    sums += numbers
ave = sums/100
print("Average of 1 - 100 =", ave)
print()

#8
divisor_number = 12
for count in range(1, 13):
    if divisor_number % count == 0:
        print("Numbers divisible by 12 =", count)
print()

#9
divisor_number = 12
counter = 0
for count in range(1, 13):
    if divisor_number % count == 0:
        counter += count
print("Sum of divisible numbers of 12 =", counter)
print()

#10
word = input("Enter a word: ")
if word == word[::-1]:
    print(f"{word} is palindrome")
else:
    print(f"{word} is not a palindrome")
print()


#11
number = (input("Enter a number: "))
if number == number[:: -1]:
        print(f"{number} is palindrome")
else:
    print(f"{number} is not a palindrome")
print()
#12

num = 4 
for nums in range(1):
    print(f"Sqare root of {num} =", sqrt(4))
print()
    
#13
sentence = input("Enter a sentence: ")
words = sentence.split()
counted = len(words)
print("Number of words in the sentence =", counted)
print()


#14
counts = 0
for even_number in range(1,21):
    if even_number % 2 == 0:
        counts += even_number
print("Total of Even numbers =", counts)
print()

#15

counts = 0
for even_number in range(1,21):
    if even_number % 3 == 0:
        counts += even_number
print("Total of odd numbers =", counts)
print()


#16

#17












